src/research-study.ts

```diff
@@ -0,0 +1,285 @@
+statusMap.set("Not yet recruiting", "approved");
+statusMap.set("Recruiting", "active");
+
+// FHIR data types supporting ResearchStudy
+export interface Identifier{
+  use?: string;
+  system?: string;
+  value?: string;
+}
+
+export interface CodeableConcept {
+  coding?: {system?: string; code?: string; display?: string;}[]
+  text?: string;
+}
+
+export interface ContactDetail {
+  name?: string;
+  telecom?: Telecom[];
+}
+
+export interface Telecom {
+  system?: string;
+  value?: string;
+  use?: string;
+}
+
+export interface Arm {
+  name?: string;
+  type?: CodeableConcept;
+  description?: string;
+}
+
+export interface Objective {
+  name?: string;
+  type?: CodeableConcept;
+}
+
+export interface Reference {
+  reference?: string;
+  type?: string;
+}
+
+// FHIR resources contained within ResearchStudy
+export interface Group {
+  resourceType?: string
+  id?: string;
+  type?: string;
+  actual?: boolean;
+  text?: Narrative; // Narrative isn't currently being used because I don't know how to wrap the trial criteria and all its spacing in xhtml
+  name?: string; // right now I'm storing criteria here instead
+}
+
+export interface Location {
+  resourceType?: string
+  id?: string;
+  name?: string;
+  telecom?: Telecom[];
+  position?: {longitude?: number; latitude?: number};
+}
+
+export interface Organization {
+  resourceType?: string
+  id?: string;
+  name?: string;
+}
+
+export interface Practitioner {
+  resourceType?: string
+  id?: string;
+  name?: HumanName[];
+}
+
+// FHIR data types supporting resources contained in ResearchStudy
+export interface Narrative{
+  status: string;
+  div: string; // should actually be escaped xhtml
+}
+
+export interface HumanName {
+  use?: string;
+  text: string;
+}
+
+// ResearchStudy implementation
+export class ResearchStudy {
+  resourceType = 'ResearchStudy';
+  id?: string;
+  identifier?: Identifier[];
+  title?: string;
+  status?: string;
+  phase?: CodeableConcept;
+  category?: CodeableConcept[];
+  condition?: CodeableConcept[];
+  contact?: ContactDetail[];
+  keyword?: CodeableConcept[];
+  location?: CodeableConcept[];
+  description?: string; // Should actually be markdown
+  arm?: Arm[];
+  objective?: Objective[];
+  enrollment?: Reference[];
+  sponsor?: Reference;
+  principalInvestigator?: Reference;
+  site?: Reference[];
+  contained?: (Group | Location | Organization | Practitioner)[];
+
+  constructor(trial: TrialScopeTrial, id: number) { // the ridiculous number of if statements are to account for empties being returned from trialscope - if there's a better eay, please let me know - also empty takes many forms ("", "[]", null, etc.) and I don't actually know what they are for each trialscope attribute and I'm not sure how to find out other than trial and error
+    this.id = String(id);
+    if (trial.nctId) {
+      this.identifier = [{use: "official", system: "http://clinicaltrilas.gov", value: trial.nctId}];
```

Comment:
I think clinicaltrials.gov was misspelled here

Reply:
Thanks for catching that! Just pushed the fix